docs/circle.rst

```diff
@@ -0,0 +1,112 @@
+==================
+:mod:`pygame_geometry.Circle`
+==================
+
+.. currentmodule:: pygame_geometry
+
+.. class:: Circle
+
+    | :sl:`pygame object for representing a circle`
+    | :sg:`Circle((x, y), radius) -> Circle`
+    | :sg:`Circle(x, y, radius) -> Circle`
+
+    The `Circle` class provides many useful methods for collision / transform and intersection.
+    A `Circle` can be created from a combination of a pair of coordinates that represent
+    the center of the circle and a radius. Circles can also be created from python objects that
+    are already a `Circle` or have an attribute named "circle".
+
+    Specifically, to construct a circle you can pass the x, y, and radius values as separate
+    arguments or inside a sequence(list or tuple).
+
+    Any function that requires a Circle argument also accepts any of these
+    values as Circles, the following are all valid ways:
+    ::
+        ((x, y), radius)
+        (x, y, radius)
+
+    The Circle functions which modify the position or size return a new copy of the
+    Circle with the affected changes. The original Circle is not modified.
+    Some methods have an alternate "in-place" version that returns None but affects the
+    original Circle. These "in-place" methods are denoted with the "ip" suffix.
+
+    The Circle class has both virtual and non-virtual attributes. Non-virtual attributes
+    are attributes that are stored in the Circle object itself. Virtual attributes are the
+    result of calculations that utilize the Circle's non-virtual attributes.
+
+    Here is the list of all the attributes of the Circle class:
+
+    .. attribute:: x
+        | :sl:`x coordinate of the center of the circle`
+        | :sg:`x -> float`
```

Comment:
Any reason for these repetitive parameter types?

Reply:
They represent what type the parameter is and what type you should assign

Reply:
But yeah i should remove those.